charts/flyte-binary/templates/config-secret.yaml

```diff
@@ -0,0 +1,54 @@
+{{- if not (include "flyte-binary.configuration.externalConfiguration" .) }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "flyte-binary.configuration.configSecretName" . }}
+  namespace: {{ .Release.Namespace | quote }}
+  labels: {{- include "flyte-binary.labels" . | nindent 4 }}
+    {{- if .Values.commonLabels }}
+    {{- tpl ( .Values.commonLabels | toYaml ) . | nindent 4 }}
+    {{- end }}
+    {{- if .Values.configuration.labels }}
+    {{- tpl ( .Values.configuration.labels | toYaml ) . | nindent 4 }}
+    {{- end }}
+  annotations:
+    {{- if .Values.commonAnnotations }}
+    {{- tpl ( .Values.commonAnnotations | toYaml ) . | nindent 4 }}
+    {{- end }}
+    {{- if .Values.configuration.annotations }}
+    {{- tpl ( .Values.configuration.annotations | toYaml ) . | nindent 4 }}
+    {{- end }}
+type: Opaque
+stringData:
+  {{- if .Values.configuration.database.password }}
+  012-database-secrets.yaml: |
+    database:
+      postgres:
+        password: {{ .Values.configuration.database.password | quote }}
+  {{- end }}
+  {{- if eq "s3" .Values.configuration.storage.provider }}
+  {{- if eq "accesskey" .Values.configuration.storage.providerConfig.s3.authType }}
+  013-storage-secrets.yaml: |
+    storage:
+      stow:
+        config:
+          access_key_id: {{ required "Access key required for S3 storage provider" .Values.configuration.storage.providerConfig.s3.accessKey | quote }}
+          secret_key: {{ required "Secret key required for S3 storage provider" .Values.configuration.storage.providerConfig.s3.secretKey | quote }}
+  {{- end }}
+  {{- end }}
+  {{- if .Values.configuration.auth.enabled }}
+  {{- if .Values.configuration.auth.enableAuthServer }}
+  {{- if .Values.configuration.auth.internal.clientSecretHash }}
+  014-auth-secrets.yaml: |
+    auth:
+      appAuth:
+        selfAuthServer:
+          staticClients:
+            flytepropeller:
+              # Instructions to generate:
+              # https://unionai.slack.com/archives/C04HP9E6WNT/p1678308193819939
+              client_secret: {{ .Values.configuration.auth.internal.clientSecretHash | quote }}
```

Comment:
It would be preferable to have `client_id` here as well

Reply:
I don't believe those are intended to be secrets. Should they be?

Reply:
They are not, however it is inconvenient to have to configure a single credential in two separate locations, for example: you wouldn't configure `AWS_ACCESS_KEY_ID` separately to `AWS_SECRET_ACCESS_KEY`.

Reply:
This particular `Secret` is generated by the chart from config specified in `values.yaml` - for basic use cases, or when installing the chart from another framework like Terraform, this will probably suffice. For cases where you are committing a `values.yaml` to git, you are better off generating a `Secret` with sensitive configuration overrides externally, and referencing that in `.Values.configuration.inlineSecretRef`.

```
$ cat <<EOF | kubectl apply -f -
 apiVersion: v1
 kind: Secret
 metadata:
   name: flyte-binary-inline-config-secret
   namespace: flyte
 type: Opaque
 stringData:
   202-database-secrets.yaml: |
     database:
       postgres:
         password: <DB_PASSWORD>
   204-auth-secrets.yaml: |
     auth:
       appAuth:
         selfAuthServer:
           staticClients:
             flytepropeller:
               client_secret: <CLIENT_SECRET_HASH>
 EOF
 ```

Reply:
This is perfect. I know you provided on Slack already but thanks for recording here as well.

Reply:
Does the key before the `.yaml` thing matter? I see you do `202-database-secrets.yaml` in your comment but in the code it is `012-database-secrets.yaml`

Reply:
It gives us control over merge order. The internal range is 1xx, so 2xx is used as the range for externally provided values. Values that are provided in subsequent files will override any existing ones.